src/components/button/button.js

```diff
@@ -54,34 +54,76 @@ function MdButtonDirective($mdInkRipple, $mdTheming, $mdAria) {
     link: postLink
   };
 
-  function isAnchor(attr) {
-    return angular.isDefined(attr.href) || angular.isDefined(attr.ngHref);
-  }
-  
-  function getTemplate(element, attr) {
-    return isAnchor(attr) ?
-           '<a class="md-button" ng-transclude></a>' :
-           '<button class="md-button" ng-transclude></button>';
-  }
-
   function postLink(scope, element, attr) {
     var node = element[0];
+    var elementHasText = node.textContent.trim();
```

Comment:
I still don't like this approach of checking if an element has text  - since it may get its text later.  We should reconsider this in the future.